cargo-quickinstall/src/lib.rs

```diff
@@ -182,17 +180,23 @@ fn untar(tarball: Vec<u8>) -> Result<String, InstallError> {
     Ok(stdout + &stderr)
 }
 
-fn curl_head(url: &str) -> Result<Vec<u8>, InstallError> {
-    let output = std::process::Command::new("curl")
-        .arg("--user-agent")
+fn prepare_curl_head_cmd(url: &str) -> std::process::Command {
+    let mut cmd = std::process::Command::new("curl");
+
+    cmd.arg("--user-agent")
         .arg("cargo-quickinstall client (alsuren@gmail.com)")
         .arg("--head")
         .arg("--silent")
         .arg("--show-error")
         .arg("--fail")
```

Comment:
Out of interest: what happens if the stats call fails when it has been spawned (you could test this by turning off WiFi or hacking your /etc/hosts to point the stats ending somewhere bogus). Does init log anything about it anywhere or is it just a tree falling in the woods with nobody listening?

Also, I just sent you an email about the stats server because this PR reminded me.

Reply:
> Out of interest: what happens if the stats call fails when it has been spawned (you could test this by turning off WiFi or hacking your /etc/hosts to point the stats ending somewhere bogus). Does init log anything about it anywhere or is it just a tree falling in the woods with nobody listening?

I believe the init process would just `wait` for it so that it can clean up the zombies, it doesn't and can't log anything as it does not have access to its stdout or stderr.

> Also, I just sent you an email about the stats server because this PR reminded me.

Thanks, I received the email.
I would reply after dinner.